web-sender/src/test/java/org/openmrs/eip/web/sender/SenderSyncArchiveControllerTest.java

```diff
@@ -0,0 +1,70 @@
+package org.openmrs.eip.web.sender;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.openmrs.eip.app.SyncConstants.MGT_DATASOURCE_NAME;
+import static org.openmrs.eip.app.SyncConstants.MGT_TX_MGR;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.openmrs.eip.app.management.entity.sender.SenderSyncArchive;
+import org.openmrs.eip.app.sender.BaseSenderTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+
+@Sql(scripts = "classpath:mgt_sender_sync_archive.sql", config = @SqlConfig(dataSource = MGT_DATASOURCE_NAME, transactionManager = MGT_TX_MGR))
+public class SenderSyncArchiveControllerTest extends BaseSenderTest {
+	
+	@Autowired
+	private SenderSyncArchiveController controller;
+	
+	@Test
+	public void shouldGetAllArchivesMessages() {
+		Map result = controller.getAll();
+		assertEquals(2, result.size());
+		assertEquals(3, result.get("count"));
+		assertEquals(3, ((List) result.get("items")).size());
+		assertEquals("e2bc25aa-1d5f-11e0-b929-000c29ad1d07",
+		    ((List<SenderSyncArchive>) result.get("items")).get(0).getIdentifier());
```

Comment:
Why only checking the first one? Might as well not check any, checking the count and item size should be enough I guess

Reply:
Done